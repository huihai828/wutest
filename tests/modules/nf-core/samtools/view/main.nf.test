nextflow_process {

    name "Test Process SAMTOOLS_VIEW"
    script "modules/nf-core/samtools/view/main.nf"
    process "SAMTOOLS_VIEW"
    profile "docker"

    test("Should run without failures") {

        when {
            params {
                // define parameters here
                outdir = "$outputDir"
            }
            process {
                """
                // define inputs of the process here
                input[0] = [
                            [ id: 'test'],
                            [ file("assets/data/sorted_mt.bam")],
                            [ file("assets/data/sorted_mt.bam.bai")]
                        ]
                input[1] = [[:], []]
                input[2] = []
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }
}