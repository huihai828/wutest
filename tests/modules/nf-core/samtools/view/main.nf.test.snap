{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.bam:md5,bdb7e02a8b3957560ab55f507663388b"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    "versions.yml:md5,277cc8b339782d658bdf35259c702710"
                ],
                "bai": [
                    
                ],
                "bam": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.bam:md5,bdb7e02a8b3957560ab55f507663388b"
                    ]
                ],
                "crai": [
                    
                ],
                "cram": [
                    
                ],
                "csi": [
                    
                ],
                "sam": [
                    
                ],
                "versions": [
                    "versions.yml:md5,277cc8b339782d658bdf35259c702710"
                ]
            }
        ],
        "timestamp": "2023-11-21T12:00:59+0000"
    }
}